pkg/block/s3/adapter.go

```diff
@@ -109,6 +116,29 @@ func NewAdapter(s3 s3iface.S3API, opts ...func(a *Adapter)) *Adapter {
 	return a
 }
 
+func (a *Adapter) getBucketRegion(ctx context.Context, bucket string) string {
+	if region, hasRegion := a.regionByBucket[bucket]; hasRegion {
+		return region
+	}
+	a.log(ctx).WithField("bucket", bucket).Debug("requesting region for bucket")
+	region, err := s3manager.GetBucketRegion(ctx, a.awsSession, bucket, "")
+	if err != nil {
+		a.log(ctx).WithError(err).Error("failed to get region for bucket, falling back to default region")
+		region = *a.awsSession.Config.Region
+	}
+	a.regionByBucket[bucket] = region
+	return region
+}
+
+func (a *Adapter) s3Client(ctx context.Context, bucket string) s3iface.S3API {
```

Comment:
are you planning to add unit tests for this piece? I mean to test that clients are created lazily.

Reply:
On it